src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[fromFormat.length - 1];
+  const newSeparator = toFormat[toFormat.length - 1];
+  const dateParts = date.split(`${oldSeparator}`);
+  const dateObject = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObject[fromFormat[i]] = dateParts[i];
+  }
+
+  if (dateObject.YY) {
+    const year = Number(dateObject.YY);
+
+    dateObject.YYYY = year < 30 ? '20' + dateObject.YY : '19' + dateObject.YY;
+  }
+
+  if (dateObject.YYYY) {
+    dateObject.YY = dateObject.YYYY.slice(-2);
+  }
+
+  const newDateArr = [];
+
+  for (let i = 0; i < toFormat.length - 1; i++) {
+    const key = toFormat[i];
+
+    if (dateObject[key]) {
+      newDateArr.push(dateObject[key]);
```

Comment:
The condition in the if statement is not necessary. Since we are iterating over the 'toFormat' array and building the 'newDateArr' based on the keys in 'dateObject', we can assume that all keys in 'toFormat' are present in 'dateObject' due to the way we've constructed it. Thus, the if check can be removed to simplify the code.